### PR TITLE
Angular 17

### LINKED ISSUE #2090: Angular 17 support
**Describe the bug**

A peer dependency conflict with apollo-angular@5.0.2.

Using Angular version 17 (@angular/core@17.0.1), apollo-angular@5.0.2 expects a peer of Angular core version 14, 15, or 16 (@angular/core@"^14.0.0 || ^15.0.0 || ^16.0.0"). 

The apollo-angular package requires a version of Angular core that is not higher than 16.

**To Reproduce**

with Angular cli 17
ng new 
npm i apollo-angular

**Expected behavior**

npm install of Apollo-angular

### PR DESCRIPTION
**Pull request by PowerKiKi**
Fixes #2090

### CODE REVIEW

**Comment by rezoled** [comment]
FYI, updating the project to Angular 17 and running it causes tests to break.
Here is a PR with the fixed code: https://github.com/kamilkisiela/apollo-angular/pull/2094

**Comment by PowerKiKi** [comment]
@MerlinMoos, @rezoled, thank you both for your contribution. I see we all work on at the same time on a very similar solution. I am trying to consolidate everything in this PR. Feel free to let me know if you think I forgot something.

**Comment by PowerKiKi** [comment]
I am having trouble with the error [`Project target "build" not found.`](https://github.com/kamilkisiela/apollo-angular/actions/runs/6858428657/job/18649217414#step:7:13), if somebody has an idea...

**Comment by rezoled** [comment]
Did you try my PR?
https://github.com/kamilkisiela/apollo-angular/pull/2094#issuecomment-1809441674
I ran all the tests successfully on my local machine

**Comment by PowerKiKi** [comment]
Unfortunately your PR did not run successfully for all Angular versions: https://github.com/kamilkisiela/apollo-angular/actions/runs/6849193747

In this PR, I dropped support for Angular 14 , 15 and 16 because of that.

I am considering dropping schematics if those prove too time-consuming to maintain (@kamilkisiela an opinion about that ?)

**Comment by rezoled** [comment]
I removed angular 14,15,16 support, can you try to run the tests again? I'm not sure why it even failed.

File: packages/apollo-angular/schematics/install/schema.json [hunk 1]
```diff
@@ -1,9 +1,16 @@
 {
   "$schema": "http://json-schema.org/schema",
   "$id": "SchematicsApolloAngularInstall",
-  "title": "Apollo Angular Install Options Schema",
+  "title": "Apollo Angular Install Options SchemaASDSAD",
```
Comment (de-dan) [inline packages/apollo-angular/schematics/install/schema.json:4]: Was this intentional?

File: packages/apollo-angular/schematics/utils/ast.ts [hunk 2]
```diff
@@ -1,8 +1,12 @@
 import * as ts from 'typescript';
-import { SchematicsException, Tree } from '@angular-devkit/schematics';
+import {  SchematicsException, Tree } from '@angular-devkit/schematics';
```
Comment (de-dan) [inline packages/apollo-angular/schematics/utils/ast.ts:2]: whats with the linting?

File: packages/apollo-angular/schematics/install/schema.json (hunk 1 above)
Comment (PowerKiKi) [reply packages/apollo-angular/schematics/install/schema.json:4]: Thanks for the review, both things are solved in #2122

**Pull request merged**